pandas/core/indexes/datetimes.py

```diff
@@ -1134,6 +1136,8 @@ def slice_indexer(self, start=None, end=None, step=None, kind=None):
     is_year_end = wrap_field_accessor(DatetimeArrayMixin.is_year_end)
     is_leap_year = wrap_field_accessor(DatetimeArrayMixin.is_leap_year)
 
+    tz_localize = wrap_array_method(DatetimeArrayMixin.tz_localize, True)
+    tz_convert = wrap_array_method(DatetimeArrayMixin.tz_convert, True)
```

Comment:
Can you explain this change?

Reply:
These two methods previously used shallow_copy in the DatetimeArray class, so `name` was inherited automatically.  This PR avoids the use of shallow_copy in the DatetimeArray class, so we need the extra step to pin `name`.